### PR TITLE
[pytest/hash] Generate available send packet ports instead of hard coding src_ports in hash_test.py.

### PR DESCRIPTION
**Pull request by William-zx**
### Description of PR

Summary:
Made the following change:

Generate available send packet ports instead of hard coding src_ports in hash_test.py.

Sub PR of [Enhance hash test #1494](https://github.com/Azure/sonic-mgmt/pull/1494)

This PR base on the following PR:

[Rewrite hash test #1506](https://github.com/Azure/sonic-mgmt/pull/1494)

[Complete hash key ingress-port test #1509](https://github.com/Azure/sonic-mgmt/pull/1509)

### Type of change

- [] Bug fix
- [] Testbed and Framework(new/improvement)
- [x] Test case(new/improvement)

### CODE REVIEW

**Comment by lguohan** [comment]
can you rebase? i will review this one.

**Comment by William-zx** [comment]
> can you rebase? i will review this one.

Conflict resolved, please review.

**Comment by William-zx** [comment]
> can you rebase? i will review this one.

@lguohan  Conflict resolved, please review.

File: tests/fib/test_fib.py [hunk 1]
```diff
@@ -129,7 +129,28 @@ def setup_hash(self, testbed, duthost, ptfhost):
 
         g_vars['testbed_type'] = testbed['topo']['name']
         g_vars['router_mac'] = duthost.shell('sonic-cfggen -d -v \'DEVICE_METADATA.localhost.mac\'')["stdout_lines"][0].decode("utf-8")
-            
+
+        # generate available send packet ports
```
Comment (lguohan) [inline tests/fib/test_fib.py:133]: this looks like a good function. can you define a function?

get_router_interface_ports()
"""get all physical ports associated with router interface (physical router interface, port channel router interface and vlan router interface)

**Comment by lguohan** [comment]
code looks good to me. can you define a function to enumerate all ports associated with a router interface on sonic switch. I think this function can be used in other places as well.

File: tests/fib/test_fib.py (hunk 1 above)
Comment (William-zx) [reply tests/fib/test_fib.py]: done.

**Comment by William-zx** [comment]
> code looks good to me. can you define a function to enumerate all ports associated with a router interface on sonic switch. I think this function can be used in other places as well.

Still put the changes in this file, and you can move it to other file later.

**Pull request merged**